docs/cpp_docs.rst

```diff
@@ -7,15 +7,33 @@ Documenting C++ Code
    This is a preview documentation format specification.
    Software documentation should currently be written in the format described at https://confluence.lsstcorp.org/display/LDMDG/Documentation+Standards
 
```

Comment:
I'm hoping to do that in DM-7892, but AFAIK there is still be useful material on that page that needs to be ported over.